### DISCUSSION TITLE
Have you thought about adding the git-bug to a desktop version?

### DISCUSSION THREAD

**Discussion by ghost** [git-bug/git-bug#896] [Q&A] [answered]
Hi.

*Have you thought about adding the git-bug to a desktop version?*

#### idea
My idea would be when I say git-bug connect - I choose git provider which can be github, gitlab, lanchpad, jira. After choosing the provider, I choose which providers I will sync.

#### idea: "tray-icon"
![image](https://user-images.githubusercontent.com/111701513/215925857-1b3465a0-79f8-4bcb-b959-dd2be347489c.png)

#### idea: "modal"
<img src="https://user-images.githubusercontent.com/111701513/194148294-22730043-760f-4a3f-a816-c0f2765174b6.png"/>

#### why?
1. This allows me to create a config file for git-bugs synchronization.
2. This allows for validation and synchronization of git-bug providers.

#### Note/license
this idea(ui) is based on [Devzstudio/Web3UIKit](https://github.com/Devzstudio/Web3UIKit).

**Comment by MichaelMure** [comment]
So you would use OAuth from those services to kick-off your local config? If so, this would work and would be pretty much the webui with some extra UI for the setup, as I hope for the webUI to eventually support external auth.

Would you be interested to work on that?

**Reply by ghost** [reply]
Hi MichaelMure. so... thank you for feedback.

> 1. So you would use OAuth from those services to kick-off your local config? 

yes, exactly.

> 2. If so, this would work and would be pretty much the webui with some extra UI for the setup, as I hope for the webUI to eventually support external auth.

this idea is wonderful.

> 3. Would you be interested to work on that?

I'm still thinking, I work a lot with the visual part of things(ui, frontend, webdesign etc).

**Comment by MichaelMure** [comment, marked as answer]
> I'm still thinking, I work a lot with the visual part of things(ui, frontend, webdesign etc).

Well, every work on the webui (which definitely need some love) get us closer to that idea.

**Reply by ghost** [reply]
Hi MichaelMure, thank you for feedback.  

> Well, every work on the webui (which definitely need some love) get us closer to that idea.

True... but... so... do you know any framework in python, javascript or go to write a ui easily for git-bug? (I ask if you have something in mind, if you have any technical indication or recommendation that you would like to be made for ui.)

**Reply by MichaelMure** [reply]
Well, it's really just a question of bundling the webui in a desktop app with few integration, so I expect that many of those solution would work. I don't know much that space but at a glance muon looks reasonable (?), especially if git-bug's core, written in go, has to be embedded.
As for the webui itself, it's written with react, graphql and material-ui.

**Reply by MichaelMure** [reply]
Or https://wails.app/ ?

**Reply by ghost** [reply]
Hi MichaelMure. **1)** what do you think of separating two repositories - for example: `git-bug-webui`, `github-bug-cli` within a git-bug organization like: `git-bug/web-ui`, `git-bug/cli`? **2)** have you thought about creating a github organization for git-bug? (doubt out of curiosity, not feature request.)

**Reply by MichaelMure** [reply]
1) what would that achieve?
2) there is one org, I never really got to use it as it doesn''t change things much compared to the current situation

**Reply by ghost** [reply]
Hi MichaelMure.

> 1. what would that achieve? 

> 2. there is one org, I never really got to use it as it doesn''t change things much compared to the current situation

So... as I just had a curious question, thanks for the feedback.
